enzyme/Enzyme/GradientUtils.h

```diff
@@ -1732,7 +1744,19 @@ class DiffeGradientUtils : public GradientUtils {
     // if (SE.getCouldNotCompute() == S)
     //  continue;
 
-    Value *ptr = invertPointerM(origptr, BuilderM);
+    Value *ptr;
+
+    switch (mode) {
+    case DerivativeMode::ForwardMode:
+      ptr = invertPointerM(origptr, BuilderM);
+      break;
+    case DerivativeMode::ReverseModePrimal:
```

Comment:
Add an assert for primal since this method should never be called in that case.

Reply:
done